NDTensors/src/lib/TypeParameterAccessors/src/unspecify_parameters.jl

```diff
@@ -0,0 +1,45 @@
+function _unspecify_parameters(type::Type)
+  return Base.typename(type).wrapper
+end
+@generated function unspecify_parameters(type_type::Type)
+  type = parameter(type_type)
+  return _unspecify_parameters(type)
+end
+
+# Like `set_parameters` but less strict, i.e. it allows
+# setting with `TypeVar` while `set_parameters` would error.
+function new_parameters(type::Type, params)
+  return to_unionall(unspecify_parameters(type){params...})
+end
+
+function _unspecify_parameter(type::Type, pos::Int)
+  !is_parameter_specified(type, pos) && return type
+  unspecified_param = parameter(unspecify_parameters(type), pos)
```

Comment:
Does it matter that you get the `TypeVar` for specifically the position `pos`, I could imagine just grabbing the first one is sufficient. I.e. does the "name" of the typevar get written into the Type. like 
```
v = unspecify_parameter(Vector{Float32}, 2) #= Array{Float32}
v.body.parameters = svec(Float32, N)
```
I think it should say N regardless of which TypeVar you take

Reply:
I think it is important to grab the right `TypeVar`. `TypeVar`s also encode type constraints, for example:
```julia
julia> struct MyType{T<:Real,N} end

julia> MyType.body.body.parameters
svec(T<:Real, N)

julia> Base.typename(MyType).wrapper{TypeVar(:X)}.body.parameters
svec(X, N)
```
so you can see if you don't grab the right one you can lose information.